spec/ruby/core/kernel/rand_spec.rb

```diff
@@ -117,6 +117,54 @@
     end
   end
 
+  context "given an inclusive range between 0 and 1" do
+    it "returns an Integer between the two Integers" do
+      1000.times do
+        x = rand(0...1)
+        x.should be_kind_of(Integer)
+        (0...1).should include(x)
+      end
+    end
+
+    it "returns a Float if at least one side is Float" do
+      1000.times do |index|
```

Comment:
Specs should avoid taking too long, so typically `1000.times` is bad for that.
I'd say just pick a fixed seed, there is no much value to test multiple seeds.
(same for other specs)